### ISSUE TITLE
Neural Network training in DKL GP (Exact) Regression

### ISSUE THREAD

**Issue by Akella17** [cornellius-gp/gpytorch#704]
This is not an issue with the GPyTorch library but a doubt regarding its implementation. How are the gradients of the neural network computed in DKL (ExactGP) Regression? Since ExactGP is a non-parametric model, how is backprop taking place? In other words, I am not sure how the loss is backpropagated across a non-parametric layer using chain-rule.

While ExactGP does not have parameters, I understand that the hyperparameters of the GP are optimized using the loss function. In the [docs](https://gpytorch.readthedocs.io/en/latest/examples/05_Scalable_GP_Regression_Multidimensional/KISSGP_Deep_Kernel_Regression_CUDA.html), it is mentioned that the parameters of the neural network and the hyperparameters of the GP are learned in an end-to-end fashion using Type-II MLE.

**Comment by jacobrgardner** [comment]
@Akella17 Intuitively, the parameters of the NN are treated as lots and lots of hyperparameters for the GP, and the NN parameters are trained through the same loss as, e.g. the lengthscale of an RBF kernel. See [the DKL paper](https://arxiv.org/abs/1511.02222).
